plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -128,6 +127,16 @@ public static Optional<String> getFilterClause(TupleDomain<ColumnHandle> tupleDo
     private static String toPredicate(PinotColumnHandle pinotColumnHandle, Domain domain)
     {
         String predicateArgument = pinotColumnHandle.isAggregate() ? pinotColumnHandle.getExpression() : quoteIdentifier(pinotColumnHandle.getColumnName());
+        if (domain.getValues().isNone()) {
+            checkState(!domain.isNullAllowed(), "IS NULL and IS NOT NULL is not supported due to different null handling semantics. See https://docs.pinot.apache.org/developers/advanced/null-value-support");
```

Comment:
So for queries like `colA = 1 or colA IS NULL` how do we handle the null semantics ?

Reply:
Created the following issues for pinot:
https://github.com/apache/pinot/issues/10600
https://github.com/apache/pinot/issues/10601

Reply:
Found workaround, will push a pr shortly and add tests